### PR TITLE
Implement tab condition.

### PR DESCRIPTION
**Pull request by vwxyzh**
#1659

### CODE REVIEW

**Comment by superyyrrzz** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Common/Loggers/WarningCodes.cs [hunk 1]
```diff
@@ -24,6 +24,7 @@ public static class Markdown
             public const string InvalidInclude = "InvalidInclude";
             public const string InvalidCodeSnippet = "InvalidCodeSnippet";
             public const string InvalidInlineCodeSnippet = "InvalidInlineCodeSnippet";
+            public const string NoVisibleTab = "NoVisibleTab";
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Common/Loggers/WarningCodes.cs]: naming: HiddenTab?

File: src/Microsoft.DocAsCode.Dfm/DfmEngineBuilder.cs [hunk 2]
```diff
@@ -96,32 +96,81 @@ public DfmEngineBuilder(Options options, string baseDir, string templateDir, IRe
...
+                    items = items.SetItem(i, new DfmTabItemBlockToken(tab.Rule, tab.Context, tab.Id, tab.Condition, tab.Title, tab.Content, visible, tab.SourceInfo));
+                }
+            }
+            if (active == -1)
+            {
+                if (firstVisibleTab != -1)
+                {
+                    active = firstVisibleTab;
+                    selectedTabIds.Add(items[firstVisibleTab].Id);
+                    Logger.LogWarning("No visible tab.", file: token.SourceInfo.File, line: token.SourceInfo.LineNumber.ToString(), code: WarningCodes.Markdown.NoVisibleTab);
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Dfm/DfmEngineBuilder.cs]: better message?

File: src/Microsoft.DocAsCode.Dfm/Tokens/DfmTabItemBlockToken.cs [hunk 3]
```diff
@@ -21,16 +21,19 @@ public class DfmTabItemBlockToken : IMarkdownExpression, IMarkdownRewritable<Dfm
 
         public DfmTabContentBlockToken Content { get; }
 
+        public bool Visible { get; }
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Dfm/Tokens/DfmTabItemBlockToken.cs]: naming: IsVisible?

File: src/Microsoft.DocAsCode.Dfm/DfmRenderer.cs [hunk 4]
```diff
@@ -194,7 +194,12 @@ private static StringBuffer RenderTabHeaders(IMarkdownRenderer renderer, DfmTabG
             for (int i = 0; i < token.Items.Length; i++)
             {
                 var item = token.Items[i];
-                sb += "<li role=\"presentation\">\n";
+                sb += "<li role=\"presentation\"";
+                if (!item.Visible)
+                {
+                    sb += " aria-hidden=\"true\" hidden=\"hidden\"";
```
Comment (hellosnow) [inline src/Microsoft.DocAsCode.Dfm/DfmRenderer.cs]: >hidden [](start = 58, length = 6)

why not true/false?

File: src/Microsoft.DocAsCode.Common/Loggers/WarningCodes.cs (hunk 1 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Common/Loggers/WarningCodes.cs]: I mean all tab is hidden in a tab group.

File: src/Microsoft.DocAsCode.Dfm/DfmRenderer.cs (hunk 4 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Dfm/DfmRenderer.cs]: https://www.w3schools.com/tags/att_global_hidden.asp

File: src/Microsoft.DocAsCode.Dfm/Tokens/DfmTabItemBlockToken.cs (hunk 3 above)
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Dfm/Tokens/DfmTabItemBlockToken.cs]: same as https://msdn.microsoft.com/en-us/library/system.web.ui.control.visible(v=vs.110).aspx

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**